app/models/concerns/submittable/solvable.rb

```diff
@@ -1,13 +1,19 @@
 module Solvable
-  def submit_solution!(user, attributes={})
-    assignment, _ = find_assignment_and_submit! user, attributes[:content].to_mumuki_solution(language)
+  def submit_solution!(user, submission_attributes={})
+    assignment, _ = find_assignment_and_submit! user, solution_for(submission_attributes)
     try_solve_discussions(user) if assignment.solved?
     assignment
   end
 
   def run_tests!(params)
     language.run_tests!(params.merge(locale: locale, expectations: expectations, custom_expectations: custom_expectations))
   end
+
+  def solution_for(submission_attributes)
+    submission_attributes[:content]
+      .to_mumuki_solution(language)
+      .with_client_result(submission_attributes[:client_result])
```

Comment:
Why don't we just add it to `to_mumuki_solution` method?

Reply:
That is because `to_mumuki_solution` is a polymorphic method, and doing so would repeat some code.